### ISSUE TITLE
[Question]: Question generation yields part of the prompt?

### ISSUE THREAD

**Issue by ram510** [run-llama/llama_index#8908]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

Here is the code I am using to generate questions. Oftentimes I see "Avoid asking questions that require prior knowledge or assumptions." as one of the questions generated. Is there a more elegant way of controlling the behaviour.
Code
`import asyncio
from typing import Dict, List

import pandas as pd
import torch.nn.functional as F
from llama_index.evaluation.dataset_generation import DatasetGenerator
from sentence_transformers import SentenceTransformer

from llama_index import Document

class RelatedQuestionGenerator:
    def __init__(
        self, response, service_context, num: int = None, user_query: str = None
    ):
        self.response = response
        self.service_context = service_context
        self.num = num
        self.user_query = user_query

    async def generate_related_questions(self) -> Dict[str, List[str]]:
        question_gen_query = f"""You are a very helpful assistant that can create only questions based on the context.
                                Your task is to ask {self.num} simple and easy questions for an upcoming research protocol.
                                Do not ask questions related to the title of the protocol.
                                The questions should be relevant and succint.
                                Restrict the questions to the context information provided."""
        doc_questions = {}
        for i in range(len(self.response.source_nodes)):
            response_text = self.response.source_nodes[i].get_text()
            response_metadata = self.response.source_nodes[i].metadata

            documents = [
                Document(
                    text=response_text,
                    metadata=response_metadata,
                )
            ]
            dataset_generator = DatasetGenerator.from_documents(
                documents, self.service_context, question_gen_query=question_gen_query
            )
            questions = await dataset_generator.agenerate_questions_from_nodes(
                num=self.num
            )

            doc_questions[response_metadata["file_name"]] = questions
        return doc_questions

    async def calculate_cosine_similarity(self, dictionary, input_string) -> List[str]:
        emb_model = SentenceTransformer("BAAI/bge-small-en-v1.5")
        input_embeddings = emb_model.encode(input_string, convert_to_tensor=True)
        filenames = []
        questions = []
        input_strings = []
        scores = []
        f_questions = []

        for filename, question_list in dictionary.items():
            for question in question_list:
                question_embedding = emb_model.encode(question, convert_to_tensor=True)
                similarity = F.cosine_similarity(
                    input_embeddings, question_embedding, dim=0
                )
                filenames.append(filename)
                questions.append(question)
                input_strings.append(input_string)
                scores.append(similarity.item())

        data = {
            "filename": filenames,
            "questions": questions,
            "input_string": input_strings,
            "score": scores,
        }
        result_df = pd.DataFrame(data)
        result_df = result_df.sort_values(by="score", ascending=False)
        questions_series = result_df["questions"]
        if questions_series.__len__() > self.num:
            f_questions.extend(questions_series.head(2).tolist())
            f_questions.extend(questions_series.tail(2).tolist())
        else:
            f_questions.extend(questions_series.tolist())

        return f_questions

    async def get_related_questions(self):
        questions_dict = await self.generate_related_questions()
        related_questions = await self.calculate_cosine_similarity(
            questions_dict, self.user_query
        )
        return related_questions
`
output:
"suggestedPrompts": [
    "What is the primary objective of the study mentioned in the context?",
    "How will protected health information (PHI) be used in this study, and who will have access to it?",
    "Where will the data be stored during the research project, and how will it be destroyed when the project is complete?",
    **"Avoid asking questions that require prior knowledge or assumptions."**
  ]

**Comment by logan-markewich** [comment]
@ram510 I've seen this happen before -- you can modify the prompt template in the kwargs or use a different LLM
